# IMAP PART 03: Downloading Fastq from the SRA Database

Welcome to the "IMAP: Downloading Fastq from SRA Database" repository! This repository is part of the IMAP (Integrated Microbiome Analysis Pipeline) project, designed to facilitate microbiome data analysis.

## About

Downloading raw sequencing data (FASTQ files) from the Sequence Read Archive (SRA) database is a crucial step in microbiome analysis. This repository provides tools and tutorials to streamline the process of retrieving FASTQ files from the SRA database.

## Key Features

- **Understanding SRA Database**: Learn about the Sequence Read Archive (SRA) database and its role in storing raw sequencing data.
- **Downloading FASTQ Files**: Step-by-step guides and scripts for downloading FASTQ files from SRA using tools like SRA Toolkit.

## Contents

- **Understanding SRA Database**: Overview of the SRA database and its importance in microbiome research.
- **Downloading FASTQ Files**: Tutorials and code examples demonstrating how to download FASTQ files from SRA using SRA Toolkit.

## Getting Started

To get started, clone or download this repository to your local machine. Follow the instructions provided in the respective directories to access tutorials, scripts, and resources for downloading FASTQ files from the SRA database.

## Dependencies

- SRA Toolkit
- Python 3.x (if using Python scripts)
- R (latest version) (if using R scripts)

Make sure to install the necessary dependencies before running any scripts.

## Contributing

Contributions to this repository are welcome! If you have suggestions, improvements, or additional resources related to downloading FASTQ files from the SRA database, feel free to open an issue or submit a pull request.

Happy downloading and analyzing your microbiome data!


<br>

# Related Works

## IMAP GitHub Repos

| IMAP-Repo                                      | Description                                               | GH-Pages                                             |
| :---------------------------------------------- | :--------------------------------------------------------- | :-----------------------------------------------------: |
| [OVERVIEW](https://github.com/tmbuza/imap-project-overview/) | IMAP project overview                                     | [Link](https://tmbuza.github.io/imap-project-overview/) |
| [PART 01](https://github.com/tmbuza/imap-essential-software/) | Software requirement for microbiome data analysis with Snakemake workflows | [Link](https://tmbuza.github.io/imap-essential-software/) |
| [PART 02](https://github.com/tmbuza/imap-sample-metadata/) | Downloading and exploring microbiome sample metadata from SRA Database | [Link](https://tmbuza.github.io/imap-sample-metadata/) |
| [PART 03](https://github.com/tmbuza/imap-download-sra-reads/) | Downloading and filtering microbiome sequencing data from SRA database | [Link](https://tmbuza.github.io/imap-download-sra-reads/) |
| [PART 04](https://github.com/tmbuza/imap-read-quality-control/) | Quality Control of Microbiome Next Generation Sequencing Reads | [Link](https://tmbuza.github.io/imap-read-quality-control/) |
| [PART 05](https://github.com/tmbuza/imap-bioinformatics-mothur/) | Microbial profiling using MOTHUR and Snakemake workflows | [Link](https://tmbuza.github.io/imap-mothur-bioinformatics/) |
| [PART 06](https://github.com/tmbuza/imap-bioinformatics-qiime2/) | Microbial profiling using QIIME2 and Snakemake workflows | [Link](https://tmbuza.github.io/imap-qiime2-bioinformatics/) |
| [PART 07](https://github.com/tmbuza/imap-data-processing/) | Processing Output from 16S-Based microbiome bioinformatics pipelines | [Link](https://tmbuza.github.io/imap-data-preparation/) |
| [PART 08](https://github.com/tmbuza/imap-exploratory-analysis/) | Exploratory Analysis of 16S-Based Microbiome Processed Data | [Link](https://tmbuza.github.io/imap-data-exploration/) |


## Session information

For a detailed overview of the tools and versions suitable for this guide, explore the [session information](session_info.txt).

## Citation
> Please consider citing the [iMAP article](https://rdcu.be/b5iVj) if you find any part of the iMAP practical user guides helpful in your microbiome data analysis.

Buza, T. M., Tonui, T., Stomeo, F., Tiambo, C., Katani, R., Schilling, M., â€¦ Kapur, V. (2019). iMAP: An integrated bioinformatics and visualization pipeline for microbiome data analysis. BMC Bioinformatics, 20. https://doi.org/10.1186/S12859-019-2965-4

## :tada: Raise awareness
> Please help increase awareness of freely available tools for microbiome data analysis.
See [Dimensions of the iMAP article](https://badge.dimensions.ai/details/id/pub.1117740326)
