# A Practical Experience with SRA Fastq Sequences {sra-fastq}

Welcome to the third installment of the IMAP Practical User Guides. In this section, we present a hands-on demo session focusing on "Downloading Microbiome Fastq Sequences from SRA Database." As we embark on this chapter, we assume that users or learners have already deposited their data, encompassing both sequences and metadata, in the SRAâ€”an integral centralized repository for microbiome studies.

Sequencing data sourced from the SRA not only provides invaluable sequences but also upholds a standardized approach to metadata reporting. Throughout this demo, our objective is to facilitate a uniform method of retrieving data, promoting consistency in subsequent analyses.

This hands-on tutorial is meticulously crafted to guide you through the essential steps of accessing and acquiring raw sequencing data. It empowers you to seamlessly integrate SRA data into your microbiome research endeavors, whether you are a seasoned researcher or new to the realm of bioinformatics. This preface sets the stage for a comprehensive understanding of the demo, ensuring a smooth exploration of this critical stage in the IMAP journey.

Microbiome read sequencing data may be obtained from various sources, including:

* Reads obtained directly from sequencing platforms by investigators.
* Reads downloaded from the Sequence Read Archive (SRA) or the European Nucleotide Archive (ENA).
* Reads synthesized using sequencing simulators.

We chose SRA for demo purpose.
